spec/requests/e2e_spec.rb

```diff
@@ -0,0 +1,49 @@
+RSpec.describe "E2E attributes" do
+  SAMPLE_VALUES = {
+    "boolean true" => true,
+    "boolean false" => false,
+    "integer" => 1,
+    "string" => "hello world",
+    "empty hash" => {},
+    "simple hash" => { "hello" => "world" },
+    "nested hash" => { "hello" => { "world" => true, "mundo" => false } },
+  }.freeze
+
+  before do
+    stub_token_response({
+      true_subject_identifier: 42,
+      pairwise_subject_identifier: "aaabbbccc",
+      scopes: %w[test_scope_write test_scope_write_2],
+    })
+  end
+
+  context "single update endpoint" do
+    SAMPLE_VALUES.each do |name, value|
+      it "sets '#{name}'" do
+        put "/v1/attributes/test_claim", headers: token_headers, params: { value: value.to_json }
+        expect(response).to be_successful
+
+        expect_attribute_has_value "test_claim", value
+      end
+    end
+  end
```

Comment:
Should we be using the name too?

```suggestion
  context "single update endpoint" do
    SAMPLE_VALUES.each do |name, value|
      it "sets '#{name}'" do
        put "/v1/attributes/#{name}", headers: token_headers, params: { value: value.to_json }
        expect(response).to be_successful

        expect_attribute_has_value name, value
      end
    end
  end
```

Reply:
No, the `name` here is the name of the test, not the name of the attribute.  I'll update the variable names.

Reply:
Ah, I see 😉